plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -164,7 +163,7 @@ deriving newtype instance ExMemoryUsage ExMemory
 deriving newtype instance ExMemoryUsage Unique
 
 -- See https://github.com/input-output-hk/plutus/issues/1861
-instance ExMemoryUsage (Some (TypeIn uni)) where
+instance ExMemoryUsage (SomeTypeIn uni) where
```

Comment:
> The only thing that we care about are the sizes of elements of buitlin types, so I think we should be able to get rid of a lot of this.

The thing is, we probably do care about the size of the type tag as we do store it at runtime. This will be especially important once we have built-in lists and tuples as the type tag for, say, a list is a sequence of constructors of `DefaultUni`, not just a single one.